File: python/atividade2/main.py
# coding=utf-8
'''
  Este material foi criado com fins de estudos e aprendizagem,
 com o objetivo de divulgar e demonstrar os meus códigos e a
 minha evolução na área.
 
 Autor: Josue Lopes

 main.py
'''

'''
 Função que recebe um intervalo de valores,
 sendo passado o início e termino e realiza
 a contagem do começo ao fim.
'''
def conta (inicio, termino) :
    for numero in range(inicio - 1, termino + 1) :
        palindrome(numero)

'''
 Verifica se o valor passado por parâmetro é 
 número primo ou não, caso seja número primo o
 valor é apresentado na tela.
'''
def palindrome (numero) :
    valor = numero
    invertido = 0
    while (valor > 0) :
        invertido *= 10
        invertido += valor % 10
        valor /= 10
    if (numero == invertido) :
        print(numero)
  
'''
 Ao iniciar o programa é solicitado um intervalo numérico 
 para que seja apresentado os valores palindromes neste meio.
'''
print("Digite um intervalo de números para se procurar os números palíndromes")
inicio = input('Começa em: ')
termino = input('Termina em: ')
conta(inicio, termino)

File: c/atividade2/main.c
 /**
 *  Este material foi criado com fins de estudos e aprendizagem,
 * com o objetivo de divulgar e demonstrar os meus códigos e a
 * minha evolução na área.
 * 
 * Autor: Josue Lopes
 * 
 * main.c
 */

#include "palindrome.h"

/**
 * Assim que o programa é iniciado é solicitado um intervalo númerico,
 * será realizada uma varredura nos valores dentro deste intervalo
 * procurando os valores palíndromes, quando encontra um esse valor é
 * apresentado na tela.
 */
int main(void){
    int minimo, maximo;

    printf("Digite o intervalo de números ");
    scanf("%d %d", &minimo, &maximo);

    if (minimo == maximo){
        printf("Não foi digitado um intervalo de números");
    } else {
        if (maximo < minimo){
            int aux = maximo;
            maximo = minimo;
            minimo = aux;
        }
        if (minimo == 0){
            minimo++;
        }
        while (minimo <= maximo){
            if (palindromeNumero(minimo)){
                printf("%d ",minimo);
            }
            minimo++;
        }
    }
    printf("\n");
    return 1;
}